### ISSUE TITLE
Usernames different from e-mail addresses

### ISSUE THREAD

**Issue by allo-** [RainLoop/rainloop-webmail#1041]
There is no option to change the username when adding an account and in the admin panel there are only the options full mailadress or the part before the @ sign. This prevents me from adding accounts with "username" as username and "my.name@server" as address. There should be an additional field, at least when adding more accounts to a given login.

**Comment by ervee** [comment]
I don't know if I understand correctly. But there is a plug-in to map usernames to email addresses. Looks like that is what you need.

**Comment by allo-** [comment]
My problem is, that usernames are unrelated to E-Mailaddresses here and the user should be able to configure the username for an account. So automatic mapping will not work for all accounts.

**Comment by ervee** [comment]
My username is not related to my email address either. That's where the custom-login-mapping plugin comes in to play. But this is not user configurable so I think you are out of luck there.

**Comment by allo-** [comment]
I hoped maybe you can add a "username" field when adding accounts to a user account in rainloop. Maybe only when the admin user checked "allow users to enter usernames" in the admin screen.

**Comment by mpietruschka** [comment]
I use one server for serveral domains. Each domain is used for one user account. Each user is able to log in with his distinct username. It's impossible to login with e-mail-address. This is good for private environments.

The problem with rainloop:

Right now it's possible to map the usernames against one server. So the user's able to login with 'firstname.lastname'. But after successfull login this is going to result in an automatically calculated e-mail-address like 'firstname.lastname@mail.server.com'. But in fact the E-Mail-Adress is 'synonym@personaldomain.com'. Technically this can be fixed with a lot of postfix work. But enduser will still be confused by the information in the UI.

Since the username isn't related to the E-Mail-Address, I don't really know a way to solve this without mapping lists.

**Comment by allo-** [comment]
Maybe there is potential for more mapping plugins:

- regex substitution: would solve my problem
- database: query a database backend with the login given for the correct alias to the login (somewhere it needs to be stored) for @mpietruschka case
- file / tcp / ... mapping

**Comment by mpietruschka** [comment]
Yes, rainloop or any plugin would have to gather this information from Mysql. Postfix is providing a nice implementation of those things... http://www.postfix.org/MYSQL_README.html

If you look from a different angle:
In some cases Rainloop could solve this by simply allowing more than one domain with the same server.

In this case I would have to find a way to change all my usernames into their corosponding e-mail-adresses. Which they aren't now!

**Comment by mpietruschka** [comment]
The autoconfig mechanism from thunderbird might solve any problem with usernames!

Before any login attempt you could lookup the server configuration. This will return the protocoll servername AND username for the current e-mail-address.

> Given the email address "fred@example.com", Thunderbird first checks <http://autoconfig.example.com/mail/config-v1.1.xml?emailaddress=fred@example.com> and then <http://example.com/.well-known/autoconfig/mail/config-v1.1.xml>.

[https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Autoconfiguration/FileFormat/HowTo#Username](url)
[https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Autoconfiguration](url)
[https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Autoconfiguration/FileFormat/HowTo](url)

**Comment by Braintelligence** [comment]
I second the notion of enabling either regular expressions in the custom mapping or allowing a user to choose his username (and maybe other settings) if the login fails. I think the latter should be easier to implement?

**Comment by RobertHeim** [comment]
here, domainname and username differ from the email name and domain. Would be great to make it configurable!
